src/commands/tileindex-validate/tileindex.validate.ts

```diff
@@ -522,3 +544,58 @@ export async function validate8BitsTiff(tiff: Tiff): Promise<void> {
     throw new Error(`${tiff.source.url.href} is not a 8 bits TIFF`);
   }
 }
+
+/**
+ * Get the list of map sheets / tiles that intersect with the given bounding box.
+ *
+ * @param bbox Bounding box of the area of interest (in EPSG:2193) to get the map sheets for (e.g. TIFF area).
+ * @param gridSize Grid size of the map sheets / tiles to get.
+ * @param minIntersectionMeters Minimum intersection area in meters (width or height) to include the map sheet.
+ */
+function getCovering(bbox: BBox, gridSize: GridSize, minIntersectionMeters = 0.15): string[] {
+  const SurroundingTiles = [
+    { x: 1, y: 0 },
+    { x: 0, y: -1 },
+    { x: -1, y: 0 },
+    { x: 0, y: 1 },
+  ];
+
+  const targetBounds = Bounds.fromBbox(bbox);
+
+  const output: string[] = [];
+  const tilesPerMapSheetSide = Math.floor(MapSheet.gridSizeMax / gridSize);
+
+  const tileWidth = Math.floor(MapSheet.width / tilesPerMapSheetSide);
+  const tileHeight = Math.floor(MapSheet.height / tilesPerMapSheetSide);
+
+  const seen = new Set();
+  const todo: Bounds[] = [];
+
+  const sheetName = getTileName(bbox[0], bbox[3], gridSize);
+  const sheetInfo = MapSheet.getMapTileIndex(sheetName);
+  if (sheetInfo == null) throw new Error('Unable to extract sheet information for point: ' + bbox[0]);
+
+  todo.push(Bounds.fromBbox(sheetInfo.bbox));
+  while (todo.length > 0) {
+    const nextBounds = todo.shift();
+    if (nextBounds == null) continue;
+
+    const nextX = nextBounds.x;
+    const nextY = nextBounds.bottom; // inverted Y axis
+    const bboxId = `${nextX}:${nextY}:${nextBounds.width}:${nextBounds.height}`;
+    // Only process each sheet once
+    if (seen.has(bboxId)) continue;
+    seen.add(bboxId);
+
+    const intersection = targetBounds.intersection(nextBounds);
+    if (intersection == null) continue; // no intersection, target mapshet is outside bounds of source
+    // Check all the surrounding tiles
+    for (const pt of SurroundingTiles) todo.push(nextBounds.add({ x: pt.x * tileWidth, y: pt.y * tileHeight })); // intersection not null, so add all neighbours
+    // Add to output only if the intersection is above the minimum coverage
+    if (intersection.width < minIntersectionMeters || intersection.height < minIntersectionMeters) continue;
+    output.push(getTileName(nextX, nextY, gridSize));
+  }
+
+  output.sort();
```

Comment:
nit: coud return as `sort()` [does also return the sorted array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#return_value)?

Reply:
done